### DISCUSSION TITLE
dvc cleanup - removing "unchecked out" cache and unchecking out data

### DISCUSSION THREAD

**Discussion by grzegorz-graczyk** [treeverse/dvc#10488] [Help]
Hi,

I am not sure if these features exist in dvc, so wanted to ask if I missed them or are these indeed not implemented:

1. Ability to garbage collect entries that are not checked out. An example would be:

```bash
$ git init .
$ dvc init
$ echo 1 > 1.txt
$ echo 2 > 2.txt
$ dvc add 1.txt 2.txt
$ git add 1.txt.dvc .gitignore 2.txt.dvc
$ git commit -m 1
$ find .dvc/cache -type f
(lists two entries)
$ rm .\2.txt
$ dvc gc --?
$ find .dvc/cache -type f
(keep only entry for 1.txt)
```
An obvious challenge is that in above case this would result in data loss - but with remote repository this is a valid use case; and the desired cache state can be reached by running `dvc pull 1.txt.dvc` on a clean cache.

2. Soft `dvc destroy` / "uncheckout". I am looking for a way to restore a dvc state of clean git checkout. The main focus here is removing the files and directories checked out using dvc, as they might be located in various places of the repository. Cache and local config probably would make sense, but these are easy to remove manually.

**Comment by dberenbaum** [comment]
> with remote repository this is a valid use case; and the desired cache state can be reached by running `dvc pull 1.txt.dvc` on a clean cache.

There are no commands exactly like those you describe, but there is one to address this use case: `dvc gc --not-in-remote`, which will garbage collect data that is already on the remote.
